In [240]:
# Initialize Otter
import otter

grader = otter.Notebook("fifa_21_raw_data.ipynb")

## Week 2 - FIFA '21 Player Ratings

This notebook is created for the week 2 assignment. Here, you have a very messy and raw dataset of EA Sports' installment of their hit FIFA series - FIFA21, which was scraped from sofifa.com.

Your task in this assignment is to clean up this dataset using the `Hints` provided in the ReadMe file

In [241]:
import pandas as pd
import numpy as np
import os
%matplotlib inline
import otter
grader = otter.Notebook()

**Question 1:** Load the fifa dataset into a variable called `fifa_dataset`. Next, write a function called `check_data` to check if the `fifa_dataset` is empty or not by returning True or False.

In [242]:
fifa_dataset = pd.read_csv('fifa_21_raw_data.csv', dtype={'Hits': str}, engine='python')

fifa_dataset.head()

def check_data(fifa_dataset):
   if fifa_dataset.empty:
      return True
   else:
      return False



In [42]:
grader.check("q1")

q1 results: All test cases passed!

**Question 1:** Load the fifa dataset into a variable called `fifa_dataset`. Next, write a function called `check_data` to check if the `fifa_dataset` is empty or not by returning True or False.

**Question 2:** Remove all duplicate player `name` from the dataset. Remember, you need to leave the first occurence and remove other ones.

In [243]:
def remove_duplicate(dataset):
    return dataset.drop_duplicates(subset=['Name'], keep='first')


In [ ]:
grader.check("q2")

**Question 1:** Load the fifa dataset into a variable called `fifa_dataset`. Next, write a function called `check_data` to check if the `fifa_dataset` is empty or not by returning True or False.

**Question 3:** Remove all leading white spaces or newline "\n" in the `club` column. For example, `"\n\n\n\nFC Barcelona"` should be transformed into `"FC Barcelona"`



In [244]:
def remove_newline(dataset):
    dataset ['Club'] = dataset['Club'].str.strip()
    return dataset

In [187]:
grader.check("q3")

q3 results: All test cases passed!

**Question 4:** List the TOP 5 countries with the highest number of representation of players in the data. Create a dictionary called `top_5_country` using the country names as the key and the number of players as the value. For example,

`top_5_country` = {'Kenya': 15,
                  'Nigeria': 45,
                  'Ghana': 14,
                  'Canada': 20,
                  'USA': 88
                 }


In [245]:
top_5_country = fifa_dataset['Nationality'].value_counts().head(5)

top_5_country = top_5_country.to_dict()

print(top_5_country)

fifa_dataset = remove_duplicate(remove_newline(fifa_dataset))

fifa_dataset.describe()


{'England': 1705, 'Germany': 1195, 'Spain': 1065, 'France': 1003, 'Argentina': 943}


,ID,Age,↓OVA,POT,BOV,Attacking,Crossing,Finishing,Heading Accuracy,Short Passing,...,GK Positioning,GK Reflexes,Total Stats,Base Stats,PAC,SHO,PAS,DRI,DEF,PHY
count,17920.000000,17920.000000,17920.000000,17920.000000,17920.000000,17920.000000,17920.000000,17920.000000,17920.000000,17920.000000,...,17920.000000,17920.000000,17920.000000,17920.000000,17920.000000,17920.000000,17920.000000,17920.000000,17920.000000,17920.000000
mean,225993.853292,25.244531,65.894141,71.234821,66.919308,249.822600,49.907087,45.999442,52.092690,58.955971,...,16.220257,16.525000,1599.574219,356.584152,67.510379,53.630915,57.873549,63.043080,50.011607,64.514621
std,27165.550622,4.699357,6.981700,6.120768,6.758423,74.461916,18.177017,19.596858,17.327394,14.529349,...,17.033423,17.892335,270.371341,40.807439,10.716734,13.840644,10.094098,9.933707,16.489535,9.605797
min,41.000000,16.000000,47.000000,47.000000,48.000000,42.000000,6.000000,3.000000,5.000000,7.000000,...,2.000000,2.000000,747.000000,232.000000,25.000000,16.000000,25.000000,25.000000,12.000000,28.000000
25%,209607.500000,21.000000,61.000000,67.000000,63.000000,223.000000,38.000000,30.000000,44.000000,54.000000,...,8.000000,8.000000,1456.000000,328.000000,61.000000,44.000000,51.000000,58.000000,35.000000,58.000000
50%,231863.500000,25.000000,66.000000,71.000000,67.000000,264.000000,54.000000,50.000000,55.000000,62.000000,...,11.000000,11.000000,1632.000000,357.000000,68.000000,56.000000,58.000000,64.000000,53.000000,66.000000
75%,246456.250000,29.000000,70.000000,75.000000,71.000000,298.000000,64.000000,62.000000,64.000000,68.000000,...,14.000000,14.000000,1785.250000,385.000000,75.000000,64.000000,65.000000,70.000000,63.000000,71.000000
max,259216.000000,53.000000,93.000000,95.000000,93.000000,437.000000,94.000000,95.000000,93.000000,94.000000,...,91.000000,90.000000,2316.000000,498.000000,96.000000,93.000000,93.000000,95.000000,91.000000,91.000000


In [193]:
grader.check("q4")

q4 results: All test cases passed!

<!-- BEGIN QUESTION -->

**Question 5:** What insight can you derive from this data?

We've 17920 uique players from 18979
Our players are aged 25 on average with 16 years old being the younges and 53 being the oldest 
England seems to have the highest number of players, followed immediately by Germany and Spain

<!-- END QUESTION -->



## Submission

Make sure you have run all cells in your notebook in order before running the cell below, so that all images/graphs appear in the output. The cell below will generate a zip file for you to submit. **Please save before exporting!**

These are some submission instructions.

In [247]:
# Save your notebook first, then run this cell to export your submission.
grader.export(run_tests=True)

/Users/chukwuemekaokonkwo/Applications/IDS/ids-fifa-21-player-ratings/venv/lib/python3.14/site-packages/otter/check/notebook.py:494: UserWarning: Could not locate a PDF to include
  warnings.warn("Could not locate a PDF to include")


OSError: xelatex not found on PATH, if you have not installed xelatex you may need to do so. Find further instructions at https://nbconvert.readthedocs.io/en/latest/install.html#installing-tex.